### PR TITLE
Support managed redis

### PR DESCRIPTION
**Pull request by Furisto**
## Description
Add support for managed Redis like AWS ElastiCache.

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖 Generated by Copilot at f6bb79f</samp>

Added support for Redis authentication in server and ws-manager-bridge components. Updated installer to allow configuring Redis username and password in web app component and pass them to the server pod. Modified `newRedisClient` function and related files to handle Redis credentials.

</details>

## Related Issue(s)

Fixes ENG-739, [part of no downtime AMI upgrades](https://linear.app/gitpod/project/no-downtime-ami-upgrades-for-meta-and-services-nodes-ad464683ace4/ENG)

## How to test
- Start a workspace in the preview environment
- Check that redis integration still works
- This is only for ensuring that no regressions are introduced. The actual testing with a managed redis service can be performed once the changes in Dedicated have been implemented.

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - fo-managed-redis</li>
	<li><b>🔗 URL</b> - <a href="https://fo-managed-redis.preview.gitpod-dev.com/workspaces" target="_blank">fo-managed-redis.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - fo-managed-redis-gha.15910</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: install/installer/pkg/config/v1/experimental/experimental.go [hunk 1]
```diff
@@ -182,6 +182,12 @@ type SpiceDBConfig struct {
 	SecretRef string `json:"secretRef"`
 }
 
+type RedisConfig struct {
+	Address   string `json:"address,omitempty"`
+	Username  string `json:"username,omitempty"`
+	SecretRef string `json:"secretRef,omitempty"`
```
Comment (svenefftinge) [inline install/installer/pkg/config/v1/experimental/experimental.go:188]: why ist this called secretref here and password further down the stack? Can we align on one way to name this property?

Comment (Furisto) [reply install/installer/pkg/config/v1/experimental/experimental.go:188]: They refer to two different things. Here we refer to a Kubernetes secret that needs to be created beforehand. This secret needs to have a password entry that contains the password. Only this entry is then passed to the deployment as environment variable.

This reminded me that we also need to pass the secret to ws-manager-bridge. I forgot that :facepalm:

**Review by akosyakov — APPROVED** [review]
@Furisto It would be nice if you leave more context how it is going to be used in Dedicated. It would be continiously running instance right without any resets? I know that we have some comments on RFC but it was not clear what was decided.

We discused now that it is generally makes sense though, so approving to unblock.

**Comment by Furisto** [comment]
> @Furisto It would be good also to have preview envs to double check that the current approach is working without problems.

I tested this in a preview environment and did not see any errors. This actually broke the preview environment creation when I created the PR because one was already there.

**Comment by Furisto** [comment]
> @Furisto It would be nice if you leave more context how it is going to be used in Dedicated. It would be continiously running instance right without any resets? I know that we have some comments on RFC but it was not clear what was decided.
> 
> We discused now that it is generally makes sense though, so approving to unblock.

Yes, it would be a continuously running instance that is managed by AWS/GCP similar to what we already do in workspace clusters.

**Pull request merged**